ui/src/components/video_recorder.jsx

```diff
@@ -0,0 +1,114 @@
+/* @flow weak */
+import React from 'react';
+import MediaStreamRecorder from 'msr';
+
+export default React.createClass({
+  displayName: 'VideoRecorder',
+
+  getInitialState() {
+    return {
+      blobURL: null,
+      videoURL: null
+    };
+  },
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.videoURL !== prevState.videoURL && this.videoEl) {
+      this.videoEl.play();
+    }
+  },
+
+  uploadBlob(blob:Blob) {
+    const url = '/teachermoments/wav';
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', url);
+    xhr.onload = this.onDoneUploading;
+    xhr.onerror = this.onErrorUploading;
+    console.log("Uploading blob...");
+    console.log(blob);
+    // xhr.send(blob);
+
+    const formData = new FormData();
+    formData.append('megaBlob-blob', new File([blob], "blobname"));
+    xhr.send(formData);    
+    console.log(formData);
+  },
+
+  onRecordVideo() {
+    var mediaConstraints = {
+      audio: true,    // don't forget audio!
+      video: true     // don't forget video!
+    };
+    this.blobs = [];
+    navigator.getUserMedia(mediaConstraints, this.onMediaSuccess, this.onMediaError);
+  },
+
+  onErrorUploading(event) {
+    console.log("Error uploading");
+  },
+
+  onDoneUploading(event) {
+    console.log("Done uploading");
+  },
+
+  onStopRecording() {
+    this.mediaRecorder.stop();
+    this.stream.stop();
+    // this.mediaRecorder.save();
+    window.ConcatenateBlobs(this.blobs, this.blobs[0].type, this.onBlobsDone);
+  },
+
+  onMediaSuccess(stream) {
+    this.mediaRecorder = new MediaStreamRecorder(stream); 
+    this.mediaRecorder.mimeType = 'video/webm';
+    this.mediaRecorder.ondataavailable = this.onDataAvailable;
+    this.mediaRecorder.start(1000);
+    this.stream = stream;
+    this.setState({videoURL: URL.createObjectURL(stream)});
+    console.log("onMediaSuccess");
+  },
+
+  onBlobsDone(megaBlob) {
+
+    console.log("onBlobsDone");
+    console.log(megaBlob);
+    // var blobURL = URL.createObjectURL(megaBlob);
+    // this.uploadBlob(this.blobs[0]);
+    // this.setState({ blobURL });
+  },
+
+  onDataAvailable(blob) {
+    // POST/PUT "Blob" using FormData/XHR2
```

Comment:
This is where we were stuck - we couldn't get data out of the video element and sent up to the server.  We tried a few different things.

We initially tried to store all the blobs, concatenate them in the browser, and push them up.  Getting the bytes for blobs that were streamed worked - we put those into `this.blobs` and they had realistic byte sizes.  But no bytes got sent over the wire.

To debug, we tried to upload a single blob using FormData/XHR2, and that seemed to work from the browser perspective, but still no payload went over the wire to the server.  To debug, we tried creating a blobURL from the blob and see if we could open the local video in the local browser, which worked.  The blob also had a realistic number of bytes.

That's where we got stuck and stopped for now.